include/onnxruntime/core/framework/data_types_internal.h

```diff
@@ -341,6 +341,26 @@ class MLTypeCallDispatcherRet {
   }
 };
 
+// Version of the MLTypeDispatcher with a return type and an input type that is passed through ('carried')
+// as the first type parameter in the call to Fn when dispatching.
+// Return type of Fn must return type convertible to Ret.
+// The value of the return type will be the return value of the function for type T which was specified for execution.
+template <class Ret, typename TCarried, template <typename, typename> class Fn, typename... Types>
+class MLTypeCallDispatcherRetWithCarriedType {
+  int32_t dt_type_;
+
+ public:
+  explicit MLTypeCallDispatcherRetWithCarriedType(int32_t dt_type) noexcept : dt_type_(dt_type) {}
+
+  template <typename... Args>
+  Ret Invoke(Args&&... args) const {
```

Comment:
>Ret [](start = 2, length = 3)

Seems the user of this Dispatcher does not really need to return a Status, change to Dispatcher with no return?

Reply:
Thanks for catching this. Technically the usage of the dispatcher in Cast::SrcDispatcher should actually return the status instead of stupidly throwing it away, so it is needed once that is fixed. :) 

---
In reply to: [469548431](https://github.com/microsoft/onnxruntime/pull/4765#discussion_r469548431) [](ancestors = 469548431)

Reply:
Actually whilst it does throw it away, it's always OK. Will change to not have a return type.

---
In reply to: [469603840](https://github.com/microsoft/onnxruntime/pull/4765#discussion_r469603840) [](ancestors = 469603840,469548431)